src/util.rs

```diff
@@ -275,3 +271,51 @@ pub fn print_dix_diff(old_generation: &Path, new_generation: &Path) -> Result<()
     }
     Ok(())
 }
+
+/// Gets a path to a previlege elevation program based on what is available in the system
+/// or from the command line arguments.
+///
+/// First this function checks for a program specified by the command line arguments,
+/// checks if it exists in the `PATH` with the `which` crate and returns a Ok result
+/// with the OsString of the path to the binary or a Err result if not.
+///
+/// If no command line arguments were found, this funtion checks for the existence of
+/// common privilege elevation program names in the `PATH` using the `which` crate and
+/// returns a Ok result with the `OsString` of the path to the binary. In the case none
+/// of the checked programs are found a Err result is returned.
+///
+/// The search is done in this order:
+///
+/// 1. `doas`
+/// 1. `sudo`
+/// 1. `run0`
+/// 1. `pkexec`
+///
+/// The logic for choosing this order is that a person with doas installed is more likely
+/// to be using it as their main privilege elevation program.
+///
+/// # Returns
+///
+/// * `Result<OsString>` - The absolute path to the privilege elevation program binary or an error if a
+/// program can't be found.
+pub fn get_elevation_program() -> Result<OsString> {
+    let args = <Main as clap::Parser>::parse();
+    if let Some(path) = args.elevation_program.map(|path| which(path)) {
+        debug!(
+            ?path,
+            "privilege elevation path specified via command line argument"
+        );
+        return Ok(path?.into_os_string());
+    }
+
+    const STRATEGIES: [&str; 4] = ["doas", "sudo", "run0", "pkexec"];
```

Comment:
Could also add `su -c` or whatever it is as a last resort strategy. I'd add this to the back of the list. You'd have to modify your logic to add the command line flag if `su` though. 

I'd also put `run0` and `pkexec` ahead of `sudo` considering many that use `run0` or `pkexec` may also have `sudo` but want to refrain from using it.

Reply:
`run0` is packaged with systemd, so it will pretty much always be present on NixOS systems. `sudo` meanwhile is its own separate option that can easily be disabled, or uninstalled on other systems. `pkexec` is enabled through polkit, which is also required for `run0`. Ordering `pkexec` vs `run0` I don't know which would be best, but setting `security.sudo.enable = false` is a lot easier than trying to disable `run0` or trying to disable the wrapper for `pkexec`, removing the unwrapped version from the `PATH`, and preserving the wrapper for `polkit-agent-helper-1`.

Reply:
Alright, fair enough.